presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayUnionFunction.java

```diff
@@ -91,4 +103,28 @@ private static void appendBigintArray(Block array, AtomicBoolean containsNull, L
             }
         }
     }
+
+    private static void appendTypedArray(Block array, TypedSet typedSet)
+    {
+        for (int i = 0; i < array.getPositionCount(); i++) {
+            typedSet.add(array, i);
+        }
+    }
+
+    private static boolean[] getDistinctBooleanValues(Block block)
```

Comment:
I would return an instance:
```
private static class DistinctBooleanValues {
   private final boolean hasNull;
   private final boolean hasTrue;
   private final boolean hasFalse;
```

that would make usage on the calling site readable.

also, please move the method right below `booleanUnion`